src/rpc/mining.cpp

```diff
@@ -115,10 +115,7 @@ static UniValue generatetoaddress(const JSONRPCRequest& request)
     std::shared_ptr<CReserveScript> coinbaseScript = std::make_shared<CReserveScript>();
     coinbaseScript->reserveScript = GetScriptForDestination(destination, colorId);
 
-    std::vector<unsigned char> privkeyraw = ParseHex(request.params[2].get_str());
-    CKey cPrivKey;
-    cPrivKey.Set(privkeyraw.begin(), privkeyraw.end(), true);
-
+    CKey cPrivKey = DecodeSecret(request.params[2].get_str());
     if(!cPrivKey.size())
```

Comment:
It is more intuitive to use `cPrivKey.IsValid()` to check the validity of the private key. In addition, it should be checked if the private key corresponds to `FederationParams().GetLatestAggregatePubkey()`.

Reply:
fixed at 37cd8f627f64269901a0954934ecaa7e614a6407